src/gui/DatabaseTabWidget.cpp

```diff
@@ -387,76 +349,31 @@ bool DatabaseTabWidget::saveDatabaseAs(Database* db)
 {
     while (true) {
         DatabaseManagerStruct& dbStruct = m_dbList[db];
-        QString oldFileName;
-        if (dbStruct.saveToFilename) {
-            oldFileName = dbStruct.filePath;
+        QString oldFilePath;
+        if (dbStruct.fileInfo.exists()) {
+            oldFilePath = dbStruct.fileInfo.absoluteFilePath();
         } else {
-            oldFileName = tr("Passwords").append(".kdbx");
+            oldFilePath = QDir::toNativeSeparators(QDir::homePath() + "/" + tr("Passwords").append(".kdbx"));
         }
-        QString fileName = fileDialog()->getSaveFileName(this, tr("Save database as"),
-                                                        oldFileName, tr("KeePass 2 Database").append(" (*.kdbx)"),
+        QString newFilePath = fileDialog()->getSaveFileName(this, tr("Save database as"),
+                                                        oldFilePath, tr("KeePass 2 Database").append(" (*.kdbx)"),
```

Comment:
@droidmonkey these arguments need to be realigned.

Reply:
done